Packs/Workday/Integrations/WorkdayIAMEventsGenerator/WorkdayIAMEventsGenerator.py

```diff
@@ -970,81 +827,158 @@
 
 
 @APP.route('/', methods=['GET'])
+def get_full_reports():
+    integration_context = get_integration_context()
+    return jsonify(integration_context)
+
+
+def test_module():
+    if int(demisto.params().get('longRunningPort', '')) and demisto.params().get("longRunning"):
+        demisto.results('ok')
+    else:
+        return_error('Please make sure the long running port is filled and the long running checkbox is marked.')
+
+
 def generate_event():
     event_type = demisto.args().get('event_type')
     user_email = demisto.args().get('user_email')
+    name = demisto.args().get('name')
     if event_type == 'hire':
-        get_new_hire_reports(user_email)
+        get_new_hire_reports(user_email, name)
     elif event_type == 'update':
         get_update_report(user_email)
     elif event_type == 'rehire':
         get_rehire_report(user_email)
     elif event_type == 'terminate':
         get_terminate_report(user_email)
 
-    integration_context = get_integration_context()
-    return jsonify(integration_context)
+    return_results(f'Successfully generated the "{event_type}" event.')
 
 
-def get_new_hire_reports(user_email):
-    if not user_email:
-        set_integration_context(NEW_HIRE_REPORT)
-    else:
-        integration_context = get_integration_context()
-        reports = NEW_HIRE_REPORT['Report_Entry']
-        for report in reports:
-            email = report.get('Email_Address')
-            if email == user_email:
-                integration_context['Report_Entry'].append(report)
-                set_integration_context(integration_context)
+def get_new_hire_reports(user_email, name):
+    integration_context = get_integration_context()
+    if not name or name == '':
+        return_error('Please insert the user full name (first name and last name).')
+    try:
+        first_name = name.split()[0]
+        last_name = name.split()[1]
```

Comment:
if there are separate first name and last name, then we better have arg for each instead of splitting name arg.
replace name with `first_name` and `last_name` args